main/src/main/scala/sbt/EvaluateTask.scala

```diff
@@ -162,19 +162,23 @@ object EvaluateTask {
   import Keys.state
 
   lazy private val sharedProgress = new TaskTimings(reportOnShutdown = true)
-
-  // sbt-pgp calls this
-  private[sbt] def defaultProgress(): ExecuteProgress[Task] = ExecuteProgress.empty[Task]
-  private[sbt] def defaultProgress(currentRef: ProjectRef): ExecuteProgress[Task] =
+  def taskTimingProgress: Option[ExecuteProgress[Task]] =
     if (java.lang.Boolean.getBoolean("sbt.task.timings")) {
       if (java.lang.Boolean.getBoolean("sbt.task.timings.on.shutdown"))
-        sharedProgress
+        Some(sharedProgress)
       else
-        new TaskTimings(reportOnShutdown = false)
-    } else {
-      if (ConsoleAppender.showProgress) new TaskProgress(currentRef)
-      else ExecuteProgress.empty[Task]
-    }
+        Some(new TaskTimings(reportOnShutdown = false))
+    } else None
+
+  def taskProgress(state: State): ExecuteProgress[Task] = {
+    val extracted = Project.extract(state)
+    import extracted.currentRef
```

Comment:
I'm not sure I see the point of this import. I might even be tempted to just write this: `new TaskProgress(Project.extract(state).currentRef)`. Another side note: `Project.extract(state)` is so ubiquitous that it might make sense to add `def extracted = Project.extract(state)` to `StateOps`.